### DISCUSSION TITLE
Nonlinear Relationship Between Watts and Load Percentage

### DISCUSSION THREAD

**Discussion by scottb721** [nutdotnet/WinNUT-Client#201] [Q&A]
Not a WinNUT issue, as it's just presenting the UPS data, but why doesn't the Load % and the wattage correlate ?
When I multiply it out as it changes to get 100%, I get widely varying wattage values.
Which is the better one to measure ?
Thanks

**Comment by gbakeman** [comment]
I've had to design several edge cases for UPSs that don't provide the exact variable measurements, often times one or both of wattage and load percent measurements were missing which meant doing a small amount of math that might be introducing some error. Please upload a full log file set to Debug, as well as one or two outputs of your variables list and let's see what's going on.

**Reply by scottb721** [reply]
The debug log makes no mention of any values so not sure it's of much use.

And here is the full Variables list,
Myups (EATON/Eaton 5E 900 AU G2/01.02.0009)
battery.charge (Battery charge (percent of full)) : 100
battery.charge.low (Remaining battery level when UPS switches to LB (percent)) : 20
battery.protection (Prevent deep discharge of battery) : yes
battery.runtime (Battery runtime (seconds)) : 1692
battery.type (Battery chemistry) : PbAc
device.mfr (Description unavailable) : EATON
device.model (Description unavailable) : Eaton 5E 900 AU G2
device.serial (Description unavailable) : BLANK
device.type (Description unavailable) : ups
driver.name (Driver name) : usbhid-ups
driver.parameter.pollfreq (Description unavailable) : 30
driver.parameter.pollinterval (Description unavailable) : 2
driver.parameter.port (Description unavailable) : auto
driver.parameter.synchronous (Description unavailable) : auto
driver.version (Driver version - NUT release) : 2.8.0
driver.version.data (Description unavailable) : MGE HID 1.46
driver.version.internal (Internal driver version) : 0.47
driver.version.usb (USB library version) : libusb-1.0.26 (API: 0x1000109)
input.voltage (Input voltage (V)) : 237.0
outlet.1.status (Outlet switch status) : on
outlet.desc (Outlet description) : Main Outlet
outlet.id (Outlet system identifier) : 0
outlet.switchable (Outlet switch ability) : no
output.frequency (Output frequency (Hz)) : 49.7
output.frequency.nominal (Nominal output frequency (Hz)) : 50
output.voltage (Output voltage (V)) : 241.0
output.voltage.nominal (Nominal output voltage (V)) : 240
ups.beeper.status (UPS beeper status) : enabled
ups.delay.shutdown (Interval to wait after shutdown with delay command (seconds)) : 20
ups.delay.start (Interval to wait before (re)starting the load (seconds)) : 30
ups.firmware (UPS firmware) : 01.02.0009
ups.load (Load on UPS (percent of full)) : 14
ups.mfr (UPS manufacturer) : EATON
ups.model (UPS model) : Eaton 5E 900 AU G2
ups.power (Current value of apparent power (VA)) : 118
ups.power.nominal (UPS power rating (VA)) : 900
ups.productid (Product ID for USB devices) : ffff
ups.realpower (Current value of real power (W)) : 90
ups.realpower.nominal (UPS real power rating (W)) : 480
ups.serial (UPS serial number) : BLANK
ups.start.battery (Allow to start UPS from battery) : yes
ups.status (UPS status) : OL
ups.timer.shutdown (Time before the load will be shutdown (seconds)) : -1
ups.timer.start (Time before the load will be started (seconds)) : -1
ups.vendorid (Vendor ID for USB devices) : 0463

**Reply by gbakeman** [reply]
Thank you for the variables, it does look like your UPS is providing a good amount so chances are WinNUT is reading them directly without any math. All the same, it would be helpful if you attach a copy of your log.

**Reply by scottb721** [reply]
Yes, NUT in Home Assistant states the same values.

The debug log just keeps repeating this ,  

WinNUT Client - 2.3
Connected
On Line
Battery OK
9/12/2024 10:41:34 PM [20324, WinNUT]: Battery Status => Charged
9/12/2024 10:41:35 PM [20324, UPS_Device]: Enter Retrieve_UPS_Datas
9/12/2024 10:41:35 PM [20324, UPS_Device]: battery.voltage is not supported by server.
9/12/2024 10:41:35 PM [20324, UPS_Device]: Apply Fallback Value when retrieving battery.voltage
9/12/2024 10:41:35 PM [20324, UPS_Device]: input.frequency is not supported by server.
9/12/2024 10:41:35 PM [20324, UPS_Device]: Apply Fallback Value when retrieving input.frequency
9/12/2024 10:41:35 PM [20324, UPS_Device]: UPS statuses have not changed since last update, skipping.
9/12/2024 10:41:35 PM [20324, WinNUT]: Updating UPS data for Form.
9/12/2024 10:41:35 PM [20324, WinNUT]: Updating battery icons based on charge percent: 100%
9/12/2024 10:41:35 PM [20324, WinNUT]: Calculated estimated remaining battery time: 00:35:06
9/12/2024 10:41:35 PM [20324, WinNUT]: Update Dial
9/12/2024 10:41:35 PM [20324, WinNUT]: Update Icon
9/12/2024 10:41:35 PM [20324, WinNUT]: NotifyIcon Text =>

**Reply by gbakeman** [reply]
> Yes, NUT in Home Assistant states the same values.

Oh OK, I understand now why you said this wasn't a WinNUT issue. 😀

> The debug log just keeps repeating this ,
> ...

That's normal for debug output during normal operation of WinNUT, I was more concerned about the beginning of the log since that's where WinNUT is making important decisions like how to interpret data from the UPS. However, since you confirmed that this isn't unique to WinNUT, I don't think that'll be necessary.

We can likely limit the culprit then to either the UPS's driver in the NUT server (usbhid-ups) or the UPS its self. It looks like you're running a version of the NUT server that's one or two minor versions behind so you may want to try testing an updated build just to see if that fixes anything. That being said, I'm trying to find your UPS on the [NUT hardware compatibility list](https://networkupstools.org/stable-hcl.html) and I'm not finding an exact match. If my suggestion above doesn't help, you may want to follow the instructions at that website and reach out to the community there to see if they have any ideas. Please post back here if you do!

Sorry I wasn't more helpful here, I hope you're able to find a solution.

**Reply by scottb721** [reply]
Appreciate the input. I'm not too concerned, was more out of curiosity. 
I have a Home Assistant power meter on the UPS output and whilst it gives a different value again, overlaying the two wattage graphs shows a general consistency, particularly in the average value.
